Import CSP report-hash and reporting tests

yoavweiss opened the pull request:
#### e7ac4c6ab702bebe9afea72a6e459ef2349d4e55
<pre>
Import CSP report-hash and reporting tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=284811">https://bugs.webkit.org/show_bug.cgi?id=284811</a>

Reviewed by Ryan Reno.

This PR imports the report-hash tests, as well as the reporting tests
that they rely on.
It currently marks these tests as timing out, as the feature is not yet
implemented.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/b9ff3b3521376a66d4765bd2765928c549d3df09">https://github.com/web-platform-tests/wpt/commit/b9ff3b3521376a66d4765bd2765928c549d3df09</a>

* LayoutTests/TestExpectations: Mark the report-hash tests as timing
  out.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/default-src.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/default-src.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/default-src.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/default-src.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/multiple-policies.https.sub-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/multiple-policies.https.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/multiple-policies.https.sub.html.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-default-src.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-default-src.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-default-src.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-default-src.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-elem.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-elem.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-elem.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-elem.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-none.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-none.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-none.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-none.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/resources/report-hash-test-runner.sub.js: Added.
(find_server_timing):
(reporting_observer_setup):
(async check_reports):
(report_hash_test):
(async no_report_test):
(async run_tests):
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/resources/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-elem.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-elem.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-elem.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-elem.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-sha512.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-sha512.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-sha512.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-sha512.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/reporting/cross-origin-reports-isolated.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/reporting/cross-origin-reports-isolated.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/cross-origin-same-site-credentials.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-bypass-report-to.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-bypass-report-to.https.sub.html.sub.headers:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-named-endpoints.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-named-endpoints.https.sub.html.sub.headers:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-override-endpoint.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-override-endpoint.https.sub.html.sub.headers:
* LayoutTests/imported/w3c/web-platform-tests/reporting/generateTestReport-honors-endpoint.http.sub-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/reporting/generateTestReport-honors-endpoint.http.sub.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/reporting/generateTestReport-honors-endpoint.http.sub.html.sub.headers: Removed.
* LayoutTests/imported/w3c/web-platform-tests/reporting/reporting-api-honors-limits.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/comment.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/comment.js.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/middle-frame.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/report-helper.js:
(checkReportExists):
(getReport):
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/same-origin-report.https.sub.html.sub.headers:
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/reporting/same-origin-cross-site-credentials.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/same-origin-same-site-credentials.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/w3c-import.log:
* LayoutTests/platform/glib/TestExpectations: Removed expectation for
  removed file.
* LayoutTests/platform/gtk/TestExpectations: Removed expectation for
  removed file.
* LayoutTests/tests-options.json:

Canonical link: <a href="https://commits.webkit.org/288157@main">https://commits.webkit.org/288157@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7e1ddb207e1c81f6386025f0fd7342cce82e22e3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/81901 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1428 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35857 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86458 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/32905 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1462 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9251 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63879 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21604 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/84971 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1081 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74534 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44163 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/981 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28711 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31352 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72324 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29320 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87889 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9143 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/6597 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72239 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9328 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70353 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71461 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17837 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15556 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14482 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/528 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9094 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/8934 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12459 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10742 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rreno commented on LayoutTests/TestExpectations:
```diff
@@ -6185,6 +6185,9 @@ imported/w3c/web-platform-tests/import-maps/nonimport-integrity.html [ DumpJSCon
 imported/w3c/web-platform-tests/import-maps/data-driven [ Skip ]
 
+# Imported tests that are timing out because the feature is not yet implemented.
+imported/w3c/web-platform-tests/content-security-policy/report-hash [ Timeout ]
```
Timeout means that we will still run the test but it will timeout whichs adds quite a bit of wall clock time overhead to the test runner. Timeouts in WPT are frequently a source of EWS slowdowns.

We should instead mark these as `[ Skip ]` and then as patches land which implement these features we can unskip them.

annevk replied:
Given this change TextExpectations for glib and gtk no longer need to be modified (glib should also include gtk).

Not entirely sure what the policy is around checking in -expected files, but I guess it's reasonable as it will more clearly show the progression in the future.

yoavweiss replied:
They were referring to a now-removed http test, which was triggering warnings. I now removed the expectation entirely (I misread it before).

annevk commented on LayoutTests/imported/w3c/web-platform-tests/reporting/cross-origin-reports-isolated.https.sub.html:
```diff
@@ -19,7 +19,7 @@
       await new Promise(resolve => {
         const iframe = document.createElement('iframe');
-        iframe.src = "https://{{hosts[alt][]}}:{{ports[https][0]}}/reporting/resources/csp-error.https.sub.html";
+        iframe.src = "https://{{domains[www]}}:{{ports[https][0]}}/reporting/resources/csp-error.https.sub.html";
```
This caused the test to timeout. I think that might be because of WebKit's limited WPT domain support. Perhaps undoing this change is better.

yoavweiss replied:
OK, I'll revert this test's import

yoavweiss replied:
done!

yoavweiss committed:
Import CSP report-hash and reporting tests
https://bugs.webkit.org/show_bug.cgi?id=284811

Reviewed by Ryan Reno.

This PR imports the report-hash tests, as well as the reporting tests
that they rely on.
It currently marks these tests as timing out, as the feature is not yet
implemented.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/b9ff3b3521376a66d4765bd2765928c549d3df09

* LayoutTests/TestExpectations: Mark the report-hash tests as timing
  out.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/default-src.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/default-src.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/default-src.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/default-src.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/multiple-policies.https.sub-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/multiple-policies.https.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/multiple-policies.https.sub.html.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-default-src.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-default-src.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-default-src.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-default-src.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-elem.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-elem.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-elem.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-elem.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-none.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-none.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-none.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src-none.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/reportonly-script-src.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/resources/report-hash-test-runner.sub.js: Added.
(find_server_timing):
(reporting_observer_setup):
(async check_reports):
(report_hash_test):
(async no_report_test):
(async run_tests):
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/resources/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-elem.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-elem.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-elem.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-elem.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-sha512.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-sha512.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-sha512.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src-sha512.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src.https.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/script-src.https.window.js.sub.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/report-hash/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/reporting/cross-origin-reports-isolated.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/reporting/cross-origin-reports-isolated.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/cross-origin-same-site-credentials.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-bypass-report-to.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-bypass-report-to.https.sub.html.sub.headers:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-named-endpoints.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-named-endpoints.https.sub.html.sub.headers:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-override-endpoint.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/document-reporting-override-endpoint.https.sub.html.sub.headers:
* LayoutTests/imported/w3c/web-platform-tests/reporting/generateTestReport-honors-endpoint.http.sub-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/reporting/generateTestReport-honors-endpoint.http.sub.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/reporting/generateTestReport-honors-endpoint.http.sub.html.sub.headers: Removed.
* LayoutTests/imported/w3c/web-platform-tests/reporting/reporting-api-honors-limits.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/comment.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/comment.js.headers: Added.
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/middle-frame.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/report-helper.js:
(checkReportExists):
(getReport):
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/same-origin-report.https.sub.html.sub.headers:
* LayoutTests/imported/w3c/web-platform-tests/reporting/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/reporting/same-origin-cross-site-credentials.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/same-origin-same-site-credentials.https.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/reporting/w3c-import.log:
* LayoutTests/platform/glib/TestExpectations: Removed expectation for
  removed file.
* LayoutTests/platform/gtk/TestExpectations: Removed expectation for
  removed file.
* LayoutTests/tests-options.json:

Canonical link: https://commits.webkit.org/288157@main
